{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from multiprocessing import Pool, Manager, Value, Lock\n",
    "\n",
    "from tqdm._tqdm_notebook import tqdm_notebook as tqdm\n",
    "tqdm.pandas()\n",
    "\n",
    "from newsplease import NewsPlease\n",
    "from itertools import product\n",
    "import os\n",
    "import pickle\n",
    "\n",
    "from datetime import datetime\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "warnings.simplefilter('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "DATA_DIR = \"../../data/reddit/Article_data_2019/\"\n",
    "DATA_FILE = \"reddit_2019jun16tojul1_articleurls\"\n",
    "data = pd.read_pickle(DATA_DIR + DATA_FILE + \".pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_data(url):\n",
    "    with lock:\n",
    "        counter.value += 1\n",
    "    if counter.value >= 1:\n",
    "        os.write(1, \"{}\\n\".format(counter.value).encode())\n",
    "    if url not in web_data:\n",
    "        try:\n",
    "            if url is not None and not url.endswith('.pdf'):\n",
    "                news = NewsPlease.from_url(url, timeout=5)\n",
    "                web_data[url] = news.get_serializable_dict()\n",
    "        except Exception as e:\n",
    "            os.write(1, url.encode())\n",
    "            os.write(1, str(datetime.now()).encode())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dask runs out of memory for god knows what reason\n",
    "# pip install bokeh for this to auto launch\n",
    "#import dask.dataframe as dd\n",
    "#from dask.multiprocessing import get\n",
    "# import sys\n",
    "#from dask.distributed import Client\n",
    "#client = Client(memory_limit=12e9)\n",
    "# goto http://localhost:8787/status\n",
    "# ddata = dd.from_pandas(data, npartitions=4)\n",
    "# res = ddata.map_partitions(lambda df: df.apply((lambda row: get_data(row)), axis=1)).compute() \n",
    "# pd.DataFrame(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('https://www.postandcourier.com/politics/warren-buttigieg-surge-in-sc-democratic-presidential-poll-as-biden/article_0a351cee-8f77-11e9-a29c-9fe60d10303b.html', 1)\n"
     ]
    }
   ],
   "source": [
    "for x in product(data.url, (1,)):\n",
    "    print(x)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: utf-8;width=device-width\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: utf-8, ie=edge, chrome=1\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: utf-8 data-n-head=true\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: \\\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: utf-8;charset=utf-8\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n",
      "error getting summary: \n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 189, in summary\n",
      "    self._html(True)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 132, in _html\n",
      "    self.html = self._parse(self.input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/readability.py\", line 141, in _parse\n",
      "    doc, self.encoding = build_doc(input)\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/htmls.py\", line 17, in build_doc\n",
      "    encoding = get_encoding(page) or 'utf-8'\n",
      "  File \"/usr/local/lib/python3.7/site-packages/readability/encoding.py\", line 46, in get_encoding\n",
      "    page.decode(encoding)\n",
      "LookupError: unknown encoding: ><script type=\n",
      "Some characters could not be decoded, and were replaced with REPLACEMENT CHARACTER.\n"
     ]
    }
   ],
   "source": [
    "# A counter...\n",
    "counter = Value('i', 0)\n",
    "lock = Lock()\n",
    "\n",
    "# parallezied outputs\n",
    "manager = Manager()\n",
    "web_data = manager.dict()\n",
    "\n",
    "p = Pool(10)\n",
    "p.map(get_data, data.url)\n",
    "p.terminate()\n",
    "p.join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "\n",
    "data2 = web_data._getvalue()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "pickle.dump(data2, open(DATA_DIR + DATA_FILE + '_temp_dict.pkl', 'wb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5777de2dd86940f6ae704c09e30a6265",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(IntProgress(value=0, max=19553), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data['article_data'] = data.progress_apply(lambda x: data2.get(x.url, None), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "data.to_pickle(DATA_DIR + DATA_FILE +'_with_text.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>num_comments</th>\n",
       "      <th>score</th>\n",
       "      <th>subreddit</th>\n",
       "      <th>subreddit_subscribers</th>\n",
       "      <th>title</th>\n",
       "      <th>url</th>\n",
       "      <th>article_data</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>c1a95h</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "      <td>JoeBiden</td>\n",
       "      <td>1039</td>\n",
       "      <td>South Carolina dem. poll: Biden 37%, Warren 17...</td>\n",
       "      <td>https://www.postandcourier.com/politics/warren...</td>\n",
       "      <td>{'authors': ['Andy Shain Ashain Postandcourier...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>c1aut2</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>JoeBiden</td>\n",
       "      <td>1039</td>\n",
       "      <td>Fox News Poll: Democrats want a steady leader;...</td>\n",
       "      <td>https://www.foxnews.com/politics/fox-news-poll...</td>\n",
       "      <td>{'authors': ['Dana Blanton'], 'date_download':...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>c1emsd</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>JoeBiden</td>\n",
       "      <td>1040</td>\n",
       "      <td>Biden calls for equal pay for US women's socce...</td>\n",
       "      <td>https://thehill.com/homenews/campaign/448815-b...</td>\n",
       "      <td>{'authors': [], 'date_download': '2019-07-07 0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>c14017</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>Pete_Buttigieg</td>\n",
       "      <td>22630</td>\n",
       "      <td>WHY SOME CANDIDATES ARE SURGING, STALLING IN IOWA</td>\n",
       "      <td>https://iowastartingline.com/2019/06/14/why-so...</td>\n",
       "      <td>{'authors': ['Pat Rynard', 'Vicki Lewis', 'Wil...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>c14jyo</td>\n",
       "      <td>5</td>\n",
       "      <td>56</td>\n",
       "      <td>Pete_Buttigieg</td>\n",
       "      <td>22634</td>\n",
       "      <td>Mayor Buttigieg named to national security org...</td>\n",
       "      <td>https://www.wndu.com/home/headlines/Buttigieg-...</td>\n",
       "      <td>{'authors': [], 'date_download': '2019-07-07 0...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       id num_comments score       subreddit subreddit_subscribers  \\\n",
       "0  c1a95h           30    29        JoeBiden                  1039   \n",
       "1  c1aut2            3    16        JoeBiden                  1039   \n",
       "2  c1emsd            0     1        JoeBiden                  1040   \n",
       "3  c14017            6    20  Pete_Buttigieg                 22630   \n",
       "4  c14jyo            5    56  Pete_Buttigieg                 22634   \n",
       "\n",
       "                                               title  \\\n",
       "0  South Carolina dem. poll: Biden 37%, Warren 17...   \n",
       "1  Fox News Poll: Democrats want a steady leader;...   \n",
       "2  Biden calls for equal pay for US women's socce...   \n",
       "3  WHY SOME CANDIDATES ARE SURGING, STALLING IN IOWA   \n",
       "4  Mayor Buttigieg named to national security org...   \n",
       "\n",
       "                                                 url  \\\n",
       "0  https://www.postandcourier.com/politics/warren...   \n",
       "1  https://www.foxnews.com/politics/fox-news-poll...   \n",
       "2  https://thehill.com/homenews/campaign/448815-b...   \n",
       "3  https://iowastartingline.com/2019/06/14/why-so...   \n",
       "4  https://www.wndu.com/home/headlines/Buttigieg-...   \n",
       "\n",
       "                                        article_data  \n",
       "0  {'authors': ['Andy Shain Ashain Postandcourier...  \n",
       "1  {'authors': ['Dana Blanton'], 'date_download':...  \n",
       "2  {'authors': [], 'date_download': '2019-07-07 0...  \n",
       "3  {'authors': ['Pat Rynard', 'Vicki Lewis', 'Wil...  \n",
       "4  {'authors': [], 'date_download': '2019-07-07 0...  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
